code/modules/clothing/spacesuits/rig_modules/rig_modules.dm

```diff
@@ -155,3 +187,80 @@
 	rig.canremove = TRUE
 	say_to_wearer("Safety lock disabled.")
 	..()
+
+
+//EMP shield module
+/obj/item/rig_module/emp_shield
+	name = "\improper EMP dissipation module"
+	desc = "A bewilderingly complex bundle of optic fibers and silicon photonic circuitry."
+	active_power_usage = 1
+
+/obj/item/rig_module/emp_shield/emp_act(severity)
+	if(activated && rig?.cell?.use(round(300/severity, 50))) // 300/150/100 cell drain to shield the suit. It might sound bad but at least the suit is keeping itself activated.
+		return TRUE
+	return FALSE
+
+
+//Rad shield module
+/obj/item/rig_module/rad_shield
+	name = "\improper R.A.D."
+	desc = "An radiation absorption device. Its acronym and true name both convey the application of this module."
+	active_power_usage = 1
+	var/event_key
+	var/initial_suit = 0
+	var/initial_helmet = 0
+	var/max_capacity = 250
+	var/current_capacity = 0
+
+/obj/item/rig_module/rad_shield/examine_addition(mob/user)
+	var/current_status = round((current_capacity/max_capacity) * 100)
+	to_chat(user, "<span class = 'notice'>The embedded [name] capacity readout reads: <font color='[current_status <= 50 ? "green" : (current_status >= 85 ? "red" : "yellow")]'>[current_status]%</font></span>")
+
+/obj/item/rig_module/rad_shield/check_activate(requires_human=FALSE)
+	if(current_capacity >= max_capacity)
+		return FALSE
+	. = ..(requires_human)
+
+/obj/item/rig_module/rad_shield/activate()
+	if(!check_activate(TRUE))
+		return
+
+	if(rig.H)
+		initial_helmet = rig.H.armor["rad"]
+		rig.H.armor["rad"] = 100
+	initial_suit = rig.armor["rad"]
+	rig.armor["rad"] = 100
+
+	say_to_wearer("[src] enabled.")
+	event_key = rig.wearer.on_irradiate.Add(src, "absorb_rads")
+	..()
+
+/obj/item/rig_module/rad_shield/deactivate()
+	if(!check_deactivate())
+		return
+
+	if(rig?.H)
+		rig.H.armor["rad"] = initial_helmet
+	rig?.armor["rad"] = initial_suit
+
+	say_to_wearer("[src] disabled.")
+	if(event_key)
+		rig.wearer?.on_irradiate.Remove(event_key)
+		event_key = null
+	..()
+
+/obj/item/rig_module/rad_shield/proc/absorb_rads(list/arguments)
+	var/mob/user = arguments["user"]
+	var/rads = arguments["rads"]
+
+	if(!rig?.wearer == user) //Well lad.
```

Comment:
```suggestion
	if(rig?.wearer == user) //Well lad.
```

Reply:
No, no. This is correct.

Reply:
do you mean this:
```suggestion
	if(rig?.wearer != user) //Well lad.
```
because `(!rig?.wearer == user)` is the same as `(!(rig?.wearer) == user)`

Reply:
Oh!

Reply:
Thanks, byondo